src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -124,18 +127,32 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
 
             throttle = new IndexThrottle();
             this.searcherFactory = new SearchFactory(engineConfig);
+            final Tuple<Long, Long> translogId; // nextTranslogId, currentTranslogId
             try {
                 writer = createWriter();
+                indexWriter = writer;
+                translogId = loadTranslogIds(writer, translog);
             } catch (IOException e) {
                 throw new EngineCreationFailureException(shardId, "failed to create engine", e);
             }
-            indexWriter = writer;
-            manager = createSearcherManager();
+            manager = createSearcherManager(translogId.v2());
             this.searcherManager = manager;
             this.mergeSchedulerFailureListener = new FailEngineOnMergeFailure();
             this.mergeSchedulerListener = new MergeSchedulerListener();
             this.mergeScheduler.addListener(mergeSchedulerListener);
             this.mergeScheduler.addFailureListener(mergeSchedulerFailureListener);
+            final TranslogRecoveryPerformer transformer = engineConfig.getTranslogRecoveryPerformer();
```

Comment:
don't think I like that guy... once the MapperAnalyzer is gone we can fold it in!